Add rate limiting on BPF events map (#29711)
cilium-team/cilium: Documentation/operations/performance/tuning.rst

performance. The overhead of Hubble is somewhere between 1-15% depending
on your network traffic patterns and Hubble aggregation settings.

In clusters with a huge amount of network traffic, cilium-agent might spend
a significant portion of CPU time on processing monitored events when Hubble
is enabled. To prevent high CPU utilization in such a case, you can set limits
on how many events can be generated by datapath code. Two limits are possible
to configure:

* Rate limit - limits how many events on average can be generated
* Burst limit - limits the number of events that can be generated in a span of 1 second

To enable eBPF Event Rate Limiting with a rate limit of 10000 and a burst limit of 50000:

> 

Reviewer: I think we should add a warning block here that when the event map rate limit is enabled, then it will impact `cilium monitor`, Hubble observability, Hubble metrics reliability, and Hubble export as we might drop events.

performance. The overhead of Hubble is somewhere between 1-15% depending
on your network traffic patterns and Hubble aggregation settings.

In clusters with a huge amount of network traffic, cilium-agent might spend
a significant portion of CPU time on processing monitored events when Hubble
is enabled. To prevent high CPU utilization in such a case, you can set limits
on how many events can be generated by datapath code. Two limits are possible
to configure:

* Rate limit - limits how many events on average can be generated
* Burst limit - limits the number of events that can be generated in a span of 1 second

> Note: Helm configuration for BPF events map rate limiting is experimental and might
> change in the upcoming releases.

Reviewer 2: ```suggestion
.. note::

    Helm configuration for BPF events map rate limiting is experimental and might
    change in upcoming releases.
```

Author: done

> Warning: when BPF events map rate limiting is enabled, Cilium monitor,
> Hubble observability, Hubble metrics reliability and Hubble export functionalities
> might be impacted due to dropped events.

Reviewer 2: ```suggestion
.. warning::

    When BPF events map rate limiting is enabled, Cilium monitor,
    Hubble observability, Hubble metrics reliability, and Hubble export functionalities
    might be impacted due to dropped events.
```

Author: done

> To enable eBPF Event Rate Limiting with a rate limit of 10000 and a burst limit of 50000:

Reviewer 2: ```suggestion
To enable eBPF Event Rate Limiting with a rate limit of 10,000 and a burst limit of 50,000:
```

Author: done

.. tabs::

>     .. group-tab:: Helm

Reviewer 2: Would you mind providing the Cilium-CLI equivalent tab, please?

Author: done

       .. parsed-literal::

           helm install cilium |CHART_RELEASE| \\
             --namespace kube-system \\
             --set bpf.events.default.rateLimit=10000 \\
             --set bpf.events.default.burstLimit=50000

If this is not sufficient, in order to optimize for maximum performance,

> Hubble can be disabled:

Reviewer 2: Nit:

```suggestion
you can disable Hubble:
```

Author: done

However, the upper capacity limits used by the Cilium agent can be overridden
for advanced users. Please refer to the :ref:`bpf_map_limitations` guide.

> eBPF Event Rate Limiting

Reviewer 3: Please consider combining this with the "Hubble" section of these docs

Author: Sure, done.
